### PR TITLE
spec: add CLI specs for notation list/login/logout/plugin

### PR DESCRIPTION
**Pull request by yizha1**

### CODE REVIEW

File: specs/commandline/list.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# notation list
+
+## Description
+
+Use `notation list` to list all the signatures of the signing artifact.
```
Comment (binbin-li) [inline specs/commandline/list.md:5]: question: should we say `signed artifact` or `signing artifact`?

File: specs/commandline/list.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  -p, --password string   Password for registry operations (default to $NOTATION_PASSWORD if not specified)
+  -u, --username string   Username for registry operations (default to $NOTATION_USERNAME if not specified)
+```
+
+## Usage
+
+### List all the signatures of the signing container image
+
+```text
+notation list https://<registry>/<repository>:<tag>
```
Comment (binbin-li) [inline specs/commandline/list.md:29]: do we need to specify `https://`?

File: specs/commandline/list.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+Upon successful execution, the digests of signatures of signing container image are printed out.
```
Comment (binbin-li) [inline specs/commandline/list.md:32]: same question, signed or signing.

File: specs/commandline/plugin.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  notation plugin list [flags]
+
+Aliases:
+  list, ls
+
+Flags:
+  -h, --help   help for list
+
+Global Flags:
+      --plain-http   Registry access via plain HTTP
```
Comment (dtzar) [inline specs/commandline/plugin.md:39]: Do we need this? If so - does it apply to plugins?

File: specs/commandline/plugin.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Currently there is no subcommand available for plugin un-installation. Plugin publisher should provide instructions to uninstall the plugin.
+
+### List installed plugins
+
+```shell
+notation plugin list
+```
+
+Upon successful execution, a list of plugins are printed out with information of name, description, version, capabilities and errors. The capabilities show what the plugin is capable of, for example, the plugin can generate signatures or verify signatures. The information of errors indicates whether the plugin installed properly. `<nil>` of Error indicates the plugin installed successfully.
```
Comment (dtzar) [inline specs/commandline/plugin.md:58]: Suggest displaying a sample output from plugin ls.
`NAME       DESCRIPTION                                   VERSION                                                  CAPABILITIES            ERROR
azure-kv   Sign artifacts with keys in Azure Key Vault   0.3.0-alpha.1+79a19c8a85f95c3e88591ac1cfcec266c0008762   [SIGNATURE_GENERATOR]   <nil>`

**Review by dtzar — CHANGES_REQUESTED** [review]

**Comment by dtzar** [comment]
Reviewed this as a group during community call today.  Once this feedback is addressed, we can merge.

File: specs/commandline/list.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Use `notation list` to list all the signatures of the signed artifact.
+
+## Outline
+
+```text
+List all the signatures of signed artifacts
+
+Usage:
+  notation list <reference> [flags]
```
[the remark this answers is not in the document]
Comment (shizhMSFT) [reply specs/commandline/list.md:13]: In Linux/Unix, `less` can be used for pagination (e.g. `notation list $ref | less`). Details of `less` can be found in `man less`.

Generally speaking, `less` will block the write pipeline and thus force `notation` stopping query the next link returned by the referrer API.

File: specs/commandline/logout.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+Use `notation logout` to log out from an OCI registry.
+
+## Outline
+
+```text
+Log out from an OCI registry
+
+Usage:
+  notation logout <server> [flags]
```
[the remark this answers is not in the document]
Comment (yizha1) [reply specs/commandline/logout.md]: use `--all` flag to log out from all logged in registries.

File: specs/commandline/list.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (yizha1) [reply specs/commandline/list.md]: > IMO default behavior should be paginated as ORAS [referrers](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md) API is paginated. Not having pagination will result in an unnecessary load on servers when 1st returned item in the list is what the user wanted. Also, it's usually good practice to have a paginated list API.

Maybe we can create another issue to address your comments. The implementation depends on the behavior of `oras-go` library. IMO, the current spec is good enough for rc.1 release.

**Review by dtzar — APPROVED** [review]

File: specs/commandline/plugin.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
+# notation plugin
+
+## Description
+
+Use `notation plugin` to manage the lifecycle of plugins. See notation [plugin documentation](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md) for more details. The `notation plugin` command by itself performs no action. In order to operate on a plugin, one of the subcommands must be used.
```
Comment (priteshbandi) [inline specs/commandline/plugin.md:5]:
```suggestion
Use `notation plugin` to manage plugins. See notation [plugin documentation](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md) for more details. The `notation plugin` command by itself performs no action. In order to operate on a plugin, one of the subcommands must be used.
```

File: specs/commandline/plugin.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+## Outline
+
+### notation plugin
+
+```text
+Manage the lifecycle of plugins
```
Comment (priteshbandi) [inline specs/commandline/plugin.md:12]:
```suggestion
Manage the plugins
```

File: specs/commandline/plugin.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+Currently there is no subcommand available for plugin un-installation. Plugin publisher should provide instructions to uninstall the plugin.
+
+### List installed plugins
+
+```shell
+notation plugin list
+```
+
+Upon successful execution, a list of plugins are printed out with information of name, description, version, capabilities and errors. The capabilities show what the plugin is capable of, for example, the plugin can generate signatures or verify signatures. The information of errors indicates whether the plugin installed properly. `<nil>` of Error indicates the plugin installed successfully.
```
Comment (priteshbandi) [inline specs/commandline/plugin.md:57]:
```suggestion
Upon successful execution, a list of plugins are printed out with information of name, description, version, capabilities and errors. The capabilities show what the plugin is capable of, for example, the plugin can generate signatures or verify signatures. The errors column indicates whether the plugin was installed properly or not, `<nil>` indicates that the plugin installed successfully.
```

File: specs/commandline/list.md [hunk 11]
```diff
@@ -2,12 +2,23 @@
...
+
+`Tags` are mutable, but `Digests` uniquely and immutably identify an artifact. If a tag is used to identify a signed artifact, notation resolves the tag to the `digest` first.
+
+Upon successful execution, both the digest of the signed artifact and the digests of signatures associated with signed artifact are printed out as following:
+
+```shell
+<registry>/<repository>@<digest>
+└── application/vnd.cncf.notary.v2.signature
+    ├──<digest_of_signature>
+    └──<digest_of_signature>
```
Comment (priteshbandi) [inline specs/commandline/list.md:15]: nit: Upon successful signature evaluation, we will short-circuit evaluation when we find verification succeeds for some signature

**Review by priteshbandi — APPROVED** [review]
LGTM with one feedback

**Review by vaninrao10 — COMMENTED** [review]
For notation login and logout, do we also outline how to login into multiple registries and logout from multiple registries. We discussed in one of the community call.

**Comment by yizha1** [comment]
> For notation login and logout, do we also outline how to login into multiple registries and logout from multiple registries. We discussed in one of the community call.

Yes, I have added support for multiple login

File: specs/commandline/list.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (priteshbandi) [reply specs/commandline/list.md]: Tacking it in https://github.com/notaryproject/notation/issues/403 for RC1

**Review by FeynmanZhou — APPROVED** [review]

File: specs/commandline/logout.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+## Usage
+
+### Log out from an OCI registry
+
+```shell
+notation logout registry.example.com
+```
+
+## Log out from all logged in registries
```
Comment (shizhMSFT) [inline specs/commandline/logout.md:29]:
```suggestion
### Log out from all logged in registries
```

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**